packages/Input/src/Input.js

```diff
@@ -4,25 +4,76 @@ import classNames from "classnames";
 import * as constants from "@paprika/constants/lib/Constants";
 import TimesCircleIcon from "@paprika/icon/lib/TimesCircle";
 import Button from "@paprika/button";
+import useI18n from "@paprika/l10n/lib/useI18n";
 import * as types from "./types";
 import * as sc from "./Input.styles";
 
+function callAll(...fns) {
+  return (...args) => {
+    for (const fn of fns) {
+      if (fn) fn(...args);
+    }
+  };
+}
+
 const Input = React.forwardRef((props, ref) => {
-  const inputClearHandler = e => {
-    e.target.value = "";
-    props.onChange(e);
-    props.onClear();
-    e.stopPropagation();
+  const {
+    a11yText,
+    clearIcon,
+    className,
+    defaultValue,
+    icon,
+    isDisabled,
+    isReadOnly,
+    hasClearButton,
+    hasError,
+    onChange,
+    onClear,
+    size,
+    value,
+    ...moreProps
+  } = props;
+
+  const isControlled = value !== null;
+
+  const i18n = useI18n();
+
+  const [shouldShowClearButton, setShouldShowClearButton] = React.useState(
+    hasClearButton && !isDisabled && !isReadOnly && (value || defaultValue)
+  );
+
+  React.useEffect(() => {
+    setShouldShowClearButton(hasClearButton && !isDisabled && !isReadOnly && (value || defaultValue));
+  }, [hasClearButton, isDisabled, isReadOnly, value, defaultValue]);
+
+  const inputRef = React.useRef();
```

Comment:
`nip` in mostly all of our components the variable using `useRef()` start with the prefix `ref`